packages/git/src/browser/git-base-widget.ts

```diff
@@ -5,18 +5,75 @@
  * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
  */
 
-import { VirtualWidget } from "@theia/core/lib/browser";
+import { VirtualWidget, SELECTED_CLASS } from "@theia/core/lib/browser";
 import { GitFileStatus, Repository, GitFileChange } from '../common';
 import URI from "@theia/core/lib/common/uri";
 import { GitRepositoryProvider } from "./git-repository-provider";
 import { LabelProvider } from "@theia/core/lib/browser/label-provider";
+import { Message } from "@phosphor/messaging";
+import { Key } from "@theia/core/lib/browser/keys";
+import { SelectionService } from "@theia/core";
+import { ElementExt } from "@phosphor/domutils";
+import { inject, injectable, postConstruct } from "inversify";
 
-export class GitBaseWidget extends VirtualWidget {
+export interface GitFileChangeNode extends GitFileChange {
+    readonly icon: string;
+    readonly label: string;
+    readonly description: string;
+    readonly caption?: string;
+    readonly extraIconClassName?: string;
+    readonly commitSha?: string;
+    selected?: boolean;
+}
+
+export namespace GitFileChangeNode {
+    export function is(node: any): node is GitFileChangeNode {
+        return 'uri' in node && 'status' in node && 'description' in node && 'label' in node && 'icon' in node;
+    }
+}
+
+@injectable()
+export class GitBaseWidget<T extends { selected?: boolean }> extends VirtualWidget {
+
+    protected gitNodes: T[];
+    private _scrollContainer: string;
+
+    @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider;
+    @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
+    @inject(SelectionService) protected readonly selectionService: SelectionService;
+
+    @postConstruct()
```

Comment:
You should do `node.selected = true` here instead of on `selectionService.onSelectionChanged`